core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -2656,15 +2672,14 @@ object Parser {
 
       override def toString = s"CharIn($min, bitSet = ..., $ranges)"
 
-      def makeError(offset: Int): Chain[Expectation] = {
-        var result = Chain.empty[Expectation]
-        var aux = ranges.toList
-        while (aux.nonEmpty) {
-          val (s, e) = aux.head
-          result = result :+ Expectation.InRange(offset, s, e)
-          aux = aux.tail
-        }
-        result
+      def makeError(offset: Int): Eval[Chain[Expectation]] = {
+        Eval.later(
+          Chain.fromSeq(
+            ranges.toList.map { case (s, e) =>
+              Expectation.InRange(offset, s, e)
+            }
+          )
```

Comment:
@rossabaker last time I saw this https://github.com/typelevel/cats-parse/pull/62 are you seeing diff results? maybe worth keep?